exwm-layout.el

```diff
@@ -405,22 +403,27 @@ selected by `other-buffer'."
 (defun exwm-layout--on-minibuffer-setup ()
   "Refresh layout when minibuffer grows."
   (exwm--log)
-  (unless (exwm-workspace--client-p)
+  ;; Only when the minibuffer's frame is an EXWM frame.
+  ;; FIXME: would it be enough checking for workspace frames?
+  (when (exwm--terminal-p)
     (exwm--defer 0 (lambda ()
                      (when (< 1 (window-height (minibuffer-window)))
                        (exwm-layout--refresh))))))
 
 (defun exwm-layout--on-echo-area-change (&optional dirty)
   "Run when message arrives or in `echo-area-clear-hook' to refresh layout."
-  (when (and (current-message)
-             (not (exwm-workspace--client-p))
-             (or (cl-position ?\n (current-message))
-                 (> (length (current-message))
-                    (frame-width exwm-workspace--current))))
-    (exwm--log)
-    (if dirty
-        (exwm-layout--refresh)
-      (exwm--defer 0 #'exwm-layout--refresh))))
+  (let ((frame (window-frame (minibuffer-window)))
```

Comment:
This seems a little redundant. `minibuffer-window` will return the minibuffer in the current frame. Then we'll ask for that frame.

Should this be `(window-frame (active-minibuffer-window))`?

Reply:
I fixed this issue in a separate commit.  Thank you for the review!